File: clients/java/Hive/src/hive/Request.java
package hive;

public class Request {
	static class RequestParams {
		public User user = null;
		public Post post = null;
		public Comment comment = null;
		public ActionType actionType = null;
	}
	
	private RequestParams params;
	private RequestType type;
	private int retries;
	
	public Request(RequestType type, RequestParams params) {
		this(type, params, 0);
	}

	public Request(RequestType type, RequestParams params, int retries) {
		this.params = params;
		this.type = type;
	}
	
	public RequestType getRequestType() {
		return type;
	}
	
	public RequestParams getRequestParams() {
		return params;
	}
	
	public int getRetries() {
		return retries;
	}
}

File: frontend/ios/Hive/Hive/Request.swift
//
//  Request.swift
//  Hive
//
//  Created by Chuck Onwuzuruike on 10/4/18.
//  Copyright © 2018 Chuck Onwuzuruike. All rights reserved.
//

import Foundation

class Request {
    
    struct RequestParams {
        var user: User?
        var post: Post?
        var comment: Comment?
        var actionType: ActionType?
    }
    
    private(set) var params: RequestParams
    private(set) var requestType: RequestType
    private(set) var retries: Int
    
    init(type: RequestType, params: RequestParams, retries: Int) {
        self.requestType = type
        self.params = params
        self.retries = retries
    }
    
    convenience init(type: RequestType, params: RequestParams) {
        self.init(type: type, params: params, retries: 0)
    }
    
}
